client/state/themes/actions.js

```diff
@@ -60,10 +60,35 @@ export function fetchNextPage( site ) {
 	};
 }
 
-export function query( params ) {
-	return {
-		type: THEMES_QUERY,
-		params: params
+function haveQueryParamsChanged( oldQueryParams, newQueryParams ) {
```

Comment:
This function contains VERY naive logic based on my understanding of which query params the consumer could change that would affect the data we want to see.